fix(html): add missing properties pages for HTMLProgressElement (#31953)
mdn/content: files/en-us/web/api/htmlprogresselement/value/index.md

---
title: "HTMLProgressElement: value property"
short-title: value
slug: Web/API/HTMLProgressElement/value
page-type: web-api-instance-property
browser-compat: api.HTMLProgressElement.value
---

> {{APIRef("DOM")}}

Reviewer: ```suggestion
{{APIRef("HTML DOM")}}
```

> The **`HTMLProgressElement.value`** sets the current progress of the {{HTMLElement("progress")}} element.

Reviewer 2: ```suggestion
The **`HTMLProgressElement.value`** represents the current progress of the {{HTMLElement("progress")}} element.
```

> The **`HTMLProgressElement.value`** represents the current progress of the {{HTMLElement("progress")}} element.

Reviewer: ```suggestion
The **`value`** property of the {{DOMxRef("HTMLProgressElement")}} interface represents the current progress of the {{HTMLElement("progress")}} element.
```

## Value

> A floating point number. If [`max`](Web/API/HTMLProgressElement/max) value is not set on the progress bar then value ranges between 0.0 and the {{domxref("HTMLProgressElement.max")}} value.

Author: This is what I wrote in the first commit. :shrug: 
Reverted to that.

> A floating point number with default value 0.0. If [`max`](Web/API/HTMLProgressElement/max) value is not set on the progress bar then value ranges between 0.0 and 1.0. If the `max` value is set then the `value` ranges between `0` and the `max` value.

Reviewer 2: Default value has a meaning for an attribute (that can be omitted) but not here: the value is always available.

```suggestion
A floating point number. If [`max`](Web/API/HTMLProgressElement/max) value is not set on the progress bar then value ranges between 0.0 and  the {{domxref("HTMLProgressElement.max")}} value.
```

> If the `value` is not set or removed then the progress bar becomes indeterminate.

Reviewer 2: What is the value here, then? How do we removed it?

Author: Good point. We can't remove value from API objects. To keep the progress bar indeterminate simply don't assign value to it:
```js
const p = document.createElement("progress");
// doesn't work
// delete p.value;

// simply don't do following to keep it indeterminate
// p.value=0;

document.body.appendChild(p);
```
I've updated  the statement.

> A floating point number. If [`max`](Web/API/HTMLProgressElement/max) value is not set on the progress bar then value ranges between 0.0 and 1.0. If the `max` value is set then the `value` ranges between `0` and the `max` value.
> 
> If the `value` property is not set on [`HTMLProgressElement`](/en-US/docs/Web/API/HTMLProgressElement) object, then the progress bar remains indeterminate.

Reviewer: ```suggestion
A floating point number. If {{DOMxRef("HTMLProgressElement.max", "max")}} value is not set on the progress bar then value ranges between 0.0 and 1.0. If the `max` value is set then the `value` ranges between `0` and the `max` value.

If the `value` property is not set on {{DOMxRef("HTMLProgressElement")}} object, then the progress bar remains indeterminate.
```

prefer using macros if available

## Examples

### HTML

```html
Determinate Progress bar: <progress id="pBar"></progress> <span>0</span>%
<br />

> Indeterminate Progress bar: <progress></progress>

Author: Point of indeterminate bar is to not have progress. Let's keep the example simple.
